invokeai/app/services/events.py

```diff
@@ -37,57 +36,73 @@ def emit_generator_progress(
             payload=dict(
                 graph_execution_state_id=graph_execution_state_id,
                 invocation=invocation_dict,
-                source_id=source_id,
-                progress_image=progress_image,
+                source_node_id=source_node_id,
+                progress_image=progress_image.dict() if progress_image is not None else None,
                 step=step,
                 total_steps=total_steps,
+                timestamp=get_timestamp(),
```

Comment:
Why don't you just add timestamps on the client-side upon receiving messages?

Otherwise, if you really want a timestamp on every message, you may want to just modify `__emit_session_event` to add it to every message automatically, but make sure you're using UTC timestamps, and handling the time zone on the client.

Reply:
> why

we need to sort images that are grabbed from the list images endpoint (which have a created timestamp, the FS ctime), and also be able to sort images that comes thru as results.

while we're at it, may as well just add timestamps directly to the events so that the UI can have the server be the source of truth for all time things (eg the ui log viewer).

moved it to the `__emit_session_event`